app/models/proposal.rb

```diff
@@ -150,6 +150,12 @@ def existing_observation_for(user)
   def add_observer(email_or_user, adder=nil, reason=nil)
     user = find_user(email_or_user)
 
+    # this authz check is here instead of in a Policy because the Policy classes
+    # are applied to the current_user, not (as in this case) the user being acted upon.
+    if client_data && !client_data.slug_matches?(user)
+      fail Pundit::NotAuthorizedError.new("May not add observer with different client.")
```

Comment:
I myself am not exactly sure what a client is in this context, and I am wondering if the end user might have a similar confusion. Would it also make sense to say something like "May not add observer from a different organization." ?